test/topology_custom/test_topology_recovery_majority_loss.py

```diff
@@ -29,6 +32,12 @@ async def test_topology_recovery_after_majority_loss(request, manager: ManagerCl
     logging.info("Waiting until driver connects to every server")
     hosts = await wait_for_cql_and_get_hosts(cql, servers[:-2], time.time() + 60)
 
+    # We need to open a separate cql session that will not ignore zero-token nodes
+    # in order to learn their hosts.
```

Comment:
I don't get it, it's the driver that is ignoring zero-token nodes, (IIRC because their `system.peers` entries are considered "incorrect" by the driver,) is there a way to override the driver's behavior?

IIRC if you connect directly to a zero-token node, then that connection will consider this zero-token node as a host indeed, but not other zero-token nodes.

Reply:
I used the approach that was taken in https://github.com/scylladb/scylladb/commit/e027ffdffc39be7dc6e1e95a40aabb2d9e30ceb3.

> is there a way to override the driver's behavior?

@patjed41 perhaps you know the answer to this question?

Reply:
https://github.com/scylladb/scylladb/commit/e027ffdffc39be7dc6e1e95a40aabb2d9e30ceb3 is consistent with what I said:
> IIRC if you connect directly to a zero-token node, then that connection will consider this zero-token node as a host indeed, but not other zero-token nodes.

there was one zero-token node and one corresponding CQL session.

If you have multiple zero-token nodes, you need separate sessions for each --- their sessions will ignore the other zero-token nodes.

Reply:
Ok. Seems to work, though, but if it is not guaranteed to be supported then I can change.

Reply:
I think using `WhiteListRoundRobinPolicy` with more than one zero-token node doesn't work in Python driver. It may "seem to work" because the driver connects only to the first contact point and keeps sending requests only there. If you try to force the driver to send a request to the second contact point (by, for example, providing it as `host` in `run_async`), the request will fail. At least that's what happened in my experiment. Maybe there is a way to make it work, I don't know.

Reply:
Now I'm creating a separate session for both of the zero-token nodes.